webui-src/app/people/people_own_contacts.js

```diff
@@ -0,0 +1,121 @@
+const m = require('mithril');
+const rs = require('rswebui');
+const peopleUtil = require('people/people_util');
+
+const contactInfo = () => {
+  let details = {};
+
+  let avatarURI = {
+    view: () => [],
+  };
+
+  return {
+    oninit: (v) =>
+      rs.rsJsonApiRequest(
+        '/rsIdentity/getIdDetails',
+        {
+          id: v.attrs.id.mGroupId,
+        },
+        (data) => {
+          details = data.details;
+          // Creating URI during fetch because `details` is uninitialized
+          // during view run, due to request being async.
+          avatarURI = peopleUtil.createAvatarURI(data.details.mAvatar);
```

Comment:
How about we turn the `createAvatarURI` into a component? It would be much easier to manage it then. Something like:
```js
const UserAvatar = () => ({
    view: (v) => {
      const imageURI = v.attrs.avatar.mData.base64;
      return imageURI === '' ?
        [] :
        m('img.avatar', {src: 'data:image/png;base64,' + imageURI });
    },
});
```
it can be used with `m(UserAvatar, {avatar: data.details.mAvatar})` and there will be no need to create intermediate components.

Reply:
Thanks. I see that you defined the same component in multiple places, can you please move it to the utils file, so that it can be defined once, and used from multiple places?

Reply:
Updated